src/librbd/internal.cc

```diff
@@ -1644,6 +1686,15 @@ bool compare_by_name(const child_info_t& c1, const child_info_t& c2)
     }
 
     return 0;
+
+  rollback:
```

Comment:
Nit: probably no rational reason to support rollback. If it fails, you should just be forced to attempt to remove it again.